src/main/java/com/zendesk/maxwell/schema/Table.java

```diff
@@ -223,8 +223,10 @@ public void diff(List<String> diffs, Table other, String nameA, String nameB) {
 	}
 
 	public void setDefaultColumnCharsets() {
+		String newCharset = this.getCharset();
 		for ( StringColumnDef c : getStringColumns() ) {
-			c.setDefaultCharset(this.getCharset());
+			int index = c.getPos();
+			columns.replace(index, c.withDefaultCharset(newCharset));
```

Comment:
seems a bit odd to rely on the result of `getPos` here.. for that matter,  I'm not totally sure why `position` would be part of our intern'ed columns...

Reply:
Using the index is a common pattern in Table to retain column order (see addColumn(int, ColumnDef), removeColumn, renameColumn, changeColumn). Though looking deeper, it seems that the code usually uses `findIndex` and only sometimes `getPos`. I think these are identical though, a ColumnDef's `pos` always == TableColumnList index per the constructor of `ImmutableColumnList`.

On interning, the idea is to create less objects overall. Multi-tenant DBs will have the same ordered columns over and over, even small variations on some won't impact the overall savings. However, we need to intern whatever provides equality and restrict mutability to keep the contract for Interner/hashCode of never changing. We can remove `pos` from ColumnDef and always scan the column list with findIndex (to find self-position) or keep it interned.